server/src/test/java/org/elasticsearch/cluster/action/shard/ShardStateActionTests.java

```diff
@@ -578,7 +592,8 @@ public void testStartedShardEntrySerialization() throws Exception {
 
         final TransportVersion version = randomFrom(randomCompatibleVersion(random()));
```

Comment:
Can we add a serialization test verifying that we can send the entry and receive the entry to/from an older version too?

Reply:
New test added here: https://github.com/elastic/elasticsearch/pull/106252/commits/ccd11d95f16055c11fdc9ca2a3fd8a8f0b6c87d1#diff-e8639cf8d716c3052fc07cc779e4a229595fc7292d31939c2e1130ab6b4ae4aaR611